src/main/java/com/wire/xenon/tools/Logger.java

```diff
@@ -64,10 +64,25 @@ public static void error(String msg) {
     }
 
     public static void error(String format, Object... args) {
+        // check if there's an exception in the objects
+        for (Object arg : args) {
+            if (arg instanceof Throwable) {
+                // if so log it as exception instead of a common format
+                exception(format, (Throwable) arg, args);
+                // break as counter and logger is handled in the exception
+                return;
```

Comment:
why return? I would think we need to extract all instances of Throwable from `args` and log them as Exceptions and then log the rest as before, no?

Reply:
You probably want to keep it in one log record for consistency, if you look into the implementation of `exception()` -> `LOGGER.log(Level.SEVERE, String.format(message, args), throwable);` - we format all args as well - so it will produce the very same message as before in addition to adding the exception to final single record

this is here just for sake of backwards compatibility, usually you should not use this `error` to log exceptions (as there's `exception` for that)

Reply:
I see.. it just looks a bit weird.